pkg/types/verifier/verifier.go

```diff
@@ -15,27 +15,17 @@
 package verifier
 
 import (
-	"fmt"
+	"crypto"
 
-	pb "github.com/sigstore/rekor-tiles/pkg/generated/protobuf"
+	"github.com/sigstore/rekor-tiles/pkg/types/verifier/identity"
 )
 
-func Validate(v *pb.Verifier) error {
-	// TODO(cmurphy): hash/public key compatibility check (https://github.com/sigstore/rekor-tiles/issues/100)
-	publicKey := v.GetPublicKey()
-	x509Cert := v.GetX509Certificate()
-	if publicKey == nil && x509Cert == nil {
-		return fmt.Errorf("missing signature public key or X.509 certificate")
-	}
-	if publicKey != nil {
-		if len(publicKey.GetRawBytes()) == 0 {
-			return fmt.Errorf("missing public key raw bytes")
-		}
-	}
-	if x509Cert != nil {
-		if len(x509Cert.GetRawBytes()) == 0 {
-			return fmt.Errorf("missing X.509 certificate raw bytes")
-		}
-	}
-	return nil
+// Verifier represents a structure that can verify a signature, e.g. a public key or certificate
+type Verifier interface {
+	// PublicKey returns the underlying public key for signature verification
+	PublicKey() crypto.PublicKey
+	// Identities returns a list of typed keys and certificates
+	Identities() ([]identity.Identity, error)
+	// String returns a human-readable representation of the verifier, e.g. PEM-encoded
+	String() string
```

Comment:
This is only used for the tests, can we do without?

Reply:
Similar to above, this function can be used by other libraries.